apps/console/app/routes/_shell.agent.$agentSlug.branch.$branchSlug.config/route.tsx

```
@@ -0,0 +1,249 @@
+      // Remove model configuration
+      const response = await fetch(
+        `/api/v1/agents/${params.agentSlug}/branches/${params.branchSlug}/models/${alias}`,
+        {
+          method: "DELETE",
+        }
+      );
+
+      if (!response.ok) {
+        const errorText = await response.text();
+        return { error: errorText || "Failed to remove model configuration" };
+      }
+
+      return { success: true, message: "Model configuration removed successfully" };
+    } else {
+      // Update model configuration
+      const response = await fetch(
+        `/api/v1/agents/${params.agentSlug}/branches/${params.branchSlug}/models/default`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            alias,
+            modelType,
+          }),
+        }
+      );
+
+      if (!response.ok) {
+        const errorText = await response.text();
+        return { error: errorText || "Failed to update model configuration" };
+      }
+
+      return { success: true, message: "Model configuration updated successfully" };
+    }
+  } catch (error) {
+    console.error("Error updating model config:", error);
+    return { error: "An unexpected error occurred" };
+  }
+}
+
+// -------------------------
+// Component
+// -------------------------
+export default function AgentBranchConfig({ loaderData, actionData }: Route.ComponentProps) {
+  const navigation = useNavigation();
+  const [isOpen, setIsOpen] = useState(false);
+
+  const shellData = useRouteLoaderData("routes/_shell") as {
+    agents: any[];
+    activeAgent: any;
+    activeBranch: any;
+  };
+
+  const defaultModel = shellData?.activeBranch?.models?.find((m: any) => m.alias === "default");
+
+  const [form, fields] = useForm({
+    defaultValue: {
+      alias: defaultModel?.alias || "default",
+      modelType: defaultModel?.type || supportedModels[0].name,
```

You are hiding a condition that is a fundamental error in the deployment. Better let an error bubble so it's clear that something is really wrong.

---

This is what I added to address this:
Required modelType with validation; removed default to supportedModels[0].
Validated modelType against supportedModels in the action; returns an error if unsupported.
Allowed remove action to bypass modelType validation, but still requires alias.
Early-return UI when supportedModels is empty to clearly indicate a deployment misconfiguration.
Disabled Save until a valid modelType is selected.